tests/OnlineSales.Tests/Environment/TestApplication.cs

```diff
@@ -44,6 +45,19 @@ public void PopulateBulkData(dynamic bulkItems)
         }
     }
 
+    public T? Get<T>(Expression<Func<T, bool>> predicate)

```

Comment:
@WASandaruwan, this is too specific function to be placed into TestApplication class. What if the next time I want to request a list of records? Or I want some kind of join... It is better to return only the DBContext instance and the the client code can do whatever it need to do with the DBContext.